### PR TITLE
Return types

### PR DESCRIPTION
**Pull request by orklah**
Hi,

In this PR, I had to modify the code a little. Mainly adding return null; or replacing return; by return null; for methods returning types.

### CODE REVIEW

**Comment by orklah** [comment]
Mmh Psalm emit an issue that seems like a false positive. I reverted my edits in this file but the issue remains... I can't even reproduce it locally

**Comment by weirdan** [comment]
As `$inner_context` is a by-ref param of that method it does indeed look like false positive. Anyway, I have the same issue emitted in an unrelated PR, so it's not something caused by your changes.

**Comment by weirdan** [comment]
@orklah a lot of `@return`s seems redundant after these changes. Do we have a reason to keep them there?

**Comment by orklah** [comment]
I removed some of them,, but I kept some of them on purpose:
- When a parent documentation is different than child native return type, Psalm complains that the two are incoherent. Redefining the phpdoc in the child fixes the issue
- A lot of case I fixed where false|null which is more precise than ?bool in signature

**Comment by orklah** [comment]
> As `$inner_context` is a by-ref param of that method it does indeed look like false positive. Anyway, I have the same issue emitted in an unrelated PR, so it's not something caused by your changes.

oh, yeah, I see that it's on master too. Thanks for noticing

**Review by weirdan — CHANGES_REQUESTED** [review]

File: src/Psalm/Type/Atomic.php [hunk 1]
```diff
@@ -575,7 +575,7 @@ public function toNamespacedString(
         ?string $this_class,
         int $php_major_version,
         int $php_minor_version
-    );
+    ): ?string;
```
[the remark this answers is not in the document]
Comment (orklah) [reply src/Psalm/Type/Atomic.php:576]: Well spotted, Dropping the parent phpdoc is enough to make psalm stop emitting issues in childs when type don't match.

**Review by weirdan — APPROVED** [review]

**Comment by muglug** [comment]
Thanks, sorry for breaking master

**Pull request merged**